discovery/consul/consul.go

```diff
@@ -50,7 +53,9 @@ func NewConsulConfig(config interface{}) (*Consul, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &Consul{*client, false}, nil
+	upstreams := make(map[string][]*consul.ServiceEntry)
+	consul := &Consul{*client, false, sync.RWMutex{}, upstreams}
```

Comment:
I thought so to but apparently you can't mix the literal `*client` and the field names. The linter gives this a pass. I think we could do:

```go
consul := &Consul{*client}
consul.wasRegistered = false
consul.lock = sync.RWMutex{}
consul.upstreams = upstreams

return consul
```